src/rdf4cpp/rdf/query/QuadPattern.cpp

```diff
@@ -0,0 +1,39 @@
+#include "QuadPattern.hpp"
+namespace rdf4cpp::rdf::query {
+QuadPattern::operator std::string() const {
+
+    return (graph().is_iri() and ((IRI) graph()).identifier() == "")  // Graph
+                   ? std::string{}
+                   : (((std::string) graph()) + " ") +
+                             (std::string) subject() + " " +  // Subject
+                             (std::string) predicate() +      // Predicate
+                             (std::string) object() + " . ";  // Object
+}
+QuadPattern::QuadPattern(Node graph, Node subject, Node predicate, Node object) : entries_({graph, subject, predicate, object}) {}
+Node &QuadPattern::graph() { return entries_[0]; }
```

Comment:
ok, would have put the context as last, but thats just me :D
but we should rename that still to context. :thinking:

Reply:
I would rather stick with graph (or graph_name). I tried to stick as close as possible to https://www.w3.org/TR/rdf11-datasets/#each-named-graph-defines-its-own-context . 
Here they also mention a context but say that it is defined by a named graph.

Reply:
fair enough. 
I was thinking about that it is actually called context, but you're right.